homeassistant/components/mobile_app/http_api.py

```diff
@@ -0,0 +1,81 @@
+"""Provides an HTTP API for mobile_app."""
+from typing import Dict
+
+from aiohttp.web import Response, Request
+
+from homeassistant.auth.util import generate_secret
+from homeassistant.components.cloud import (async_create_cloudhook,
+                                            async_is_logged_in)
+from homeassistant.components.http import HomeAssistantView
+from homeassistant.components.http.data_validator import RequestDataValidator
+from homeassistant.const import (HTTP_CREATED, HTTP_INTERNAL_SERVER_ERROR,
+                                 CONF_WEBHOOK_ID)
+
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.storage import Store
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import (ATTR_REGISTRATIONS, ATTR_SUPPORTS_ENCRYPTION,
+                    CONF_CLOUDHOOK_ID, CONF_CLOUDHOOK_URL, CONF_SECRET,
+                    CONF_USER_ID, DOMAIN, REGISTER_DEVICE_SCHEMA)
+
+from .helpers import supports_encryption, savable_state
+
+from .webhook import setup_device
+
+
+def register_http_handlers(hass: HomeAssistantType, store: Store) -> bool:
+    """Register the HTTP handlers/views."""
+    hass.http.register_view(DevicesView(store))
+    return True
+
+
+class DevicesView(HomeAssistantView):
+    """A view that accepts device registration requests."""
+
+    url = '/api/mobile_app/devices'
+    name = 'api:mobile_app:register-device'
+
+    def __init__(self, store: Store) -> None:
+        """Initialize the view."""
+        self._store = store
+
+    @RequestDataValidator(REGISTER_DEVICE_SCHEMA)
+    async def post(self, request: Request, data: Dict) -> Response:
+        """Handle the POST request for device registration."""
+        hass = request.app['hass']
+
+        webhook_id = generate_secret()
+
+        if async_is_logged_in(hass):
+            cloudhook = await async_create_cloudhook(hass, webhook_id)
+
+            if cloudhook is not None:
+                data[CONF_CLOUDHOOK_ID] = cloudhook[CONF_CLOUDHOOK_ID]
```

Comment:
Actually thinking now, we shouldn't even expose cloudhook ID, it's an implementation detail.

Reply:
So no need to store it either.

Reply:
[We need to store it since `cloudhooks.async_create` doesn't return information about an existing cloudhook if one already exists](https://github.com/robbiet480/home-assistant/blob/78e82e68e623c89f35c74338e262fdf0b83de898/homeassistant/components/cloud/cloudhooks.py#L30-L31). We use it to determine if we should be sending the headers to tell the app to upgrade to a cloudhook.

Reply:
Cloud ID is internal. The current check if cloud is ok (async_logged_in) is also wrong, as it doesn't check subscription. I have a new method that I will put in a PR tonight. I think that instead of using a header to give a new URL, we should just have a header `coud-available: true;`, and the app will call `update_registration` hook with empty data which will include the cloudhook url if one was made?

Reply:
That sounds good! So we will just always return the cloudhook URL in `update_registration` from now one (assuming one is available).

Reply:
Update this code to not store `cloudhook_id`